src/gateway/GasService.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {Auth} from "src/Auth.sol";
+import {IAggregatorV3} from "src/interfaces/IAggregatorV3.sol";
+import {MathLib} from "src/libraries/MathLib.sol";
+import {MessagesLib} from "src/libraries/MessagesLib.sol";
+
+contract GasService is IAggregatorV3, Auth {
+    using MathLib for uint256;
+
+    uint8 public constant decimals = 18;
+    string public constant description = "CFG/ETH price feed";
+    uint256 public constant version = 1;
+
+    uint80 roundId;
+    uint256 updatedTime;
+    uint256 price;
+    uint256 messageCost;
+    uint256 proofCost;
+
+    event File(bytes32 what, uint256 value);
+
+    constructor(uint256 messageCost_, uint256 proofCost_, uint256 price_) {
+        messageCost = messageCost_;
+        proofCost = proofCost_;
+        price = price_;
+        updatedTime = block.timestamp;
+        roundId = 1;
+    }
+
+    function file(bytes32 what, uint256 value) external auth {
+        if (what == "messageCost") messageCost = value;
+        if (what == "proofCost") proofCost = value;
+        else revert("CentrifugeGasService/file-unrecognized-param");
+        emit File(what, value);
+    }
+
+    function updatePrice(uint256 value) external auth {
```

Comment:
I think we should consider actually adding a message for this. So Centrifuge Chain can periodically send messages updating the gas price directly.

Reply:
Adding a message as a message within the MessageLib and handle it ?